src/Attribute/Parameter/CollectionResolver.php

```diff
@@ -29,19 +33,77 @@ public function getParameterValue(
             return Result::fail();
         }
 
+        if (is_a($attribute->className, BackedEnum::class, true)) {
+            /**
+             * @psalm-suppress ArgumentTypeCoercion Because class name is backed enumeration name.
+             */
+            $collection = $this->createCollectionOfBackedEnums($resolvedValue, $attribute->className);
+        } else {
+            $collection = $this->createCollectionOfObjects(
+                $resolvedValue,
+                $context->getHydrator(),
+                $attribute->className
+            );
+        }
+
+        return Result::success($collection);
+    }
+
+    /**
+     * @psalm-param class-string $className
+     * @return object[]
+     */
+    private function createCollectionOfObjects(
+        iterable $resolvedValue,
+        HydratorInterface $hydrator,
+        string $className
+    ): array {
         $collection = [];
         foreach ($resolvedValue as $item) {
             if (!is_array($item) && !$item instanceof DataInterface) {
                 continue;
             }
 
             try {
-                $collection[] = $context->getHydrator()->create($attribute->className, $item);
+                $collection[] = $hydrator->create($className, $item);
             } catch (NonInstantiableException) {
                 continue;
             }
         }
+        return $collection;
+    }
 
-        return Result::success($collection);
+    /**
+     * @psalm-param class-string<BackedEnum> $className
+     * @return BackedEnum[]
+     */
+    private function createCollectionOfBackedEnums(iterable $resolvedValue, string $className): array
+    {
+        $collection = [];
+        $isStringBackedEnum = $this->isStringBackedEnum($className);
+        foreach ($resolvedValue as $item) {
+            if ($item instanceof $className) {
+                $collection[] = $item;
+                continue;
+            }
+
+            if (is_string($item) || is_int($item)) {
+                $enum = $className::tryFrom($isStringBackedEnum ? (string) $item : (int) $item);
+                if ($enum !== null) {
+                    $collection[] = $enum;
+                }
+            }
+        }
+        return $collection;
+    }
+
+    /**
+     * @psalm-param class-string<BackedEnum> $className
+     */
+    private function isStringBackedEnum(string $className): bool
```

Comment:
```suggestion
    private function isBackedEnumClass(string $className): bool
```

Reply:
This method check that class name is string typed enum.

Reply:
Makes sense then.